meeshkan/serve/mock/specs.py

```diff
@@ -0,0 +1,43 @@
+import os
+import logging
+from openapi_typed_2 import (
+    convert_to_openapi,
+    OpenAPIObject,
+)
+from typing import List
+import json
+import yaml
+from dataclasses import dataclass
+
+
+@dataclass
+class OpenAPISpecification:
+    """An OpenAPI object with information about the source it was created from."""
+
+    api: OpenAPIObject
+    source: str
+
+
+def load_specs(specs_dir: str) -> List[OpenAPISpecification]:
+    if not os.path.exists(specs_dir):
+        logging.info("OpenAPI schema directory not found %s", specs_dir)
+        return []
+    specs_with_sources = []
+    specs_paths = [
+        s
+        for s in os.listdir(specs_dir)
+        if s.endswith("yml") or s.endswith("yaml") or s.endswith("json")
+    ]
+    for spec_path in specs_paths:
+        with open(os.path.join(specs_dir, spec_path), encoding="utf8") as spec_file:
+            specs_with_sources.append(
```

Comment:
Could this maybe be split into multiple steps? It can just be a bit hard to debug if there are many possible ways it can go wrong?

Reply:
Done!